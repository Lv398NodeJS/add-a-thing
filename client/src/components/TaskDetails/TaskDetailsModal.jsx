import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import TaskDetails from './TaskDetails';
import SubtaskListContainer from '../Subtask/SubtaskListContainer';
import DeleteTask from './DeleteTask';

export default class TaskDetailsModal extends React.Component {
  constructor() {
    super();
    this.state = {
      showDelete: false,
    };
  }

  closeTaskDetails = () => {
    const { onClose: close } = this.props;
    close();
  };

  closeDeleteModal = () => {
    const { showDelete } = this.state;
    this.setState({
      showDelete: !showDelete,
    });
  };

  render() {
    const { show: modalShow } = this.props;
    const { showDelete } = this.state;
    return (
      <Modal
        show={modalShow}
        onHide={this.closeTaskDetails}
        aria-labelledby="contained-modal-title-center"
        centered
      >
        <Modal.Body>
          <TaskDetails
            closeTaskDetails={this.closeTaskDetails}
          />
          <SubtaskListContainer />
        </Modal.Body>
        <Button
          variant="outline-danger"
          onClick={() => this.setState({ showDelete: !showDelete })}
          className="delete-task-details-button float-right"
        >
          {'Delete'}
        </Button>
        <DeleteTask
          showDelete={showDelete}
          closeTaskDetails={this.closeTaskDetails}
          closeDeleteModal={this.closeDeleteModal}
        />
      </Modal>
    );
  }
}
