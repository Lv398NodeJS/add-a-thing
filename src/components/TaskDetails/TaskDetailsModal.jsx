import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import TaskDetails from './TaskDetails';
import SubTaskList from '../SubTaskList/SubTaskList';
import DeleteTask from './DeleteTask';

export default class TaskDetailsModal extends React.Component {
  constructor() {
    super();
    this.state = {
      showDelete: false,
    };
  }

  closeTaskDetails = () => {
    const { onClose: close } = this.props;
    close();
  };

  closeDeleteModal = () => {
    const { showDelete } = this.state;
    this.setState({
      showDelete: !showDelete,
    });
  };

  render() {
    const { taskRef, show: modalShow } = this.props;
    const { showDelete } = this.state;
    return (
      <Modal
        show={modalShow}
        onHide={this.closeTaskDetails}
        aria-labelledby="contained-modal-title-center"
        centered
      >
        <Modal.Body>
          <TaskDetails
            taskRef={taskRef}
            closeTaskDetails={this.closeTaskDetails}
          />
          <SubTaskList taskRef={taskRef} />
        </Modal.Body>
        <Button
          variant="outline-danger"
          onClick={() => this.setState({ showDelete: !showDelete })}
          className="delete-task-details-button float-right"
        >
          {'Delete'}
        </Button>
        <DeleteTask
          taskRef={taskRef}
          showDelete={showDelete}
          closeTaskDetails={this.closeTaskDetails}
          closeDeleteModal={this.closeDeleteModal}
        />
      </Modal>
    );
  }
}
