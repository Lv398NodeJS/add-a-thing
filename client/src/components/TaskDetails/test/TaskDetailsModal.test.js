import React from 'react';
import { shallow } from 'enzyme';
import TaskDetailsModal from '../TaskDetailsModal';

describe('TaskDetailsModal', () => {
  const component = shallow(<TaskDetailsModal />);
  it('should render Modal window correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it("open modal window when 'Delete' button clicked", () => {
    component.find('.delete-task-details-button').simulate('click');

    expect(component.state().showDelete).toEqual(true);
  });
});
