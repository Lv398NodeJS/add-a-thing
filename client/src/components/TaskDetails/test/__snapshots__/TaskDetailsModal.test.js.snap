// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetailsModal should render Modal window correctly 1`] = `ShallowWrapper {}`;
